### PR TITLE
Drop tracker geometries T22,T23,T27,T28 - first commit

### PR DESCRIPTION
**Pull request by emiglior**
#### PR description:

This PR concerns Upgrade.

This PR removes the phase-2 Tracker geometries T22, T23, T27 T28 definitively discarded after the choice of the Inner Tracker layout

ref. https://indico.cern.ch/event/1154888/contributions/4849182/attachments/2434145/4168775/news20220429.pdf slide 3

The PR removes both the geometry files and the corresponding workflows.
Furthermore, as T27 geometry is removed, the GT for geometries with bricked pixels is now 'phase2_realistic_T29'

The file
Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
is also changed in order to preserve the workflow numbers.

Geometries for which large PU200 samples were produced
https://its.cern.ch/jira/browse/PDMVMCPROD-30
are still kept.

No changes are expected in the output for the geometries/workflows which are kept.

#### PR validation:

The PR was validated with runTheMatrix checking that
- w/f numbers for geometries from 2026D76 to 2026D90 did not change
- running explicitely the w/f 36600.0 (2026D81) 39400.0 (2026D88) 40200.0 (2026D90)

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

This PR is not a backport

@adewit @connorpa @mmusich @tsusa @tvami FYI

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca
 * PR drops Phase-2 tk geometries, the symbolic GT changes accordingly
 * Jenkins tests pass, differences are known

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

This PR removes the unused tracker after decision was made on the inner tracker. No change is expected from the PR test as removed workflow has not been used for the short matrix. PR reserves the workflow number as expected.

**Comment by kskovpen** [comment]
+pdmv

File: Configuration/AlCa/python/autoCondPhase2.py [hunk 1]
```diff
@@ -28,52 +26,42 @@
...
     'T25' : ( ','.join( [ 'SiPixelSimLorentzAngle_phase2_T25_v0_mc' ,SiPixelSimLARecord,connectionString, "", "2021-03-16 20:00:00.000"] ), ), #uH = 0.053/T (TBPX L2,L3,L4), uH=0.0/T (TBPX L1 TEPX+TFPX)
     'T26' : ( ','.join( [ 'SiPixelSimLorentzAngle_phase2_T25_v0_mc' ,SiPixelSimLARecord,connectionString, "", "2021-03-16 20:00:00.000"] ), ), #uH = 0.053/T (TBPX L2,L3,L4), uH=0.0/T (TBPX L1 TEPX+TFPX)
-    'T27' : ( ','.join( [ 'SiPixelSimLorentzAngle_phase2_T25_v0_mc' ,SiPixelSimLARecord,connectionString, "", "2021-03-16 20:00:00.000"] ), ), #uH = 0.053/T (TBPX L2,L3,L4), uH=0.0/T (TBPX L1 TEPX+TFPX)
+    'T29' : ( ','.join( [ 'SiPixelSimLorentzAngle_phase2_T25_v0_mc' ,SiPixelSimLARecord,connectionString, "", "2021-03-16 20:00:00.000"] ), ), #uH = 0.053/T (TBPX L2,L3,L4), uH=0.0/T (TBPX L1 TEPX+TFPX)
     'T30' : ( ','.join( [ 'SiPixelSimLorentzAngle_phase2_IT_v6.4.0_25x100_v1_mc' ,SiPixelSimLARecord,connectionString, "", "2021-12-03 16:00:00.000"] ), ), #uH = 0.053/T (TBPX), uH=0.0/T (TEPX+TFPX)
 }
 
 allTags["GenError"] = {
```
Comment (perrotta) [inline Configuration/AlCa/python/autoCondPhase2.py:62]: Just to check that nothing escaped: isn't "T29" needed for `allTags["GenError"]` and `allTags["Template"]`

Comment (emiglior) [reply Configuration/AlCa/python/autoCondPhase2.py:62]: As T29 has few TBPX modules with bricked planar pixels, there are no 'GenError' nor 'Template' payloads for this geometry.

**Comment by emiglior** [comment]
> @emiglior if this is the "first commit", what we should expect from the "second" one? Aren't geometries T22,T23,T27,T28 already fully dropped with this PR?

@perrotta Indeed T22,T23,T27,T28 are fully dropped with this PR. 
"first commit"  it is a bad wording in the commit message... just to express that I was not confident that one iteration was enough

**Comment by perrotta** [comment]
+1

**Pull request merged**